tests/unit/utils.unitTest.ts

```diff
@@ -46,5 +46,63 @@ describe("utils", () => {
 
       expect(filteredRecords.length).toBe(0);
     });
+
+    it("should filter correctly events that have IVA With Certificate, ivaDefects populated but not test result fail", async () => {
+      expandedRecords[0].testTypes.testTypeClassification = "IVA With Certificate";
```

Comment:
I think it would be better to have a function inside of the describe that you could use to take a deep clone of the json test data, without actually changing the value **just incase** that it causes any issues with other tests due to the test data being directly changed. There's also no need to have these as async functions due to there being no awaits inside of them

Reply:
Good spot on the async, I was just replicating what was already there.

I've repushed with these now removed for my tests and existing tests.

Reply:
> Before each test it is resetting expandedRecords so should never have that problem.
> 
> ```
>   beforeEach(() => {
>     expandedRecords = StreamService.getTestResultStream(event);
>   });
> ```

Isn't that re importing the stream event every time each of the tests are ran? I think that it would be better and most likely faster to only import it once and do a deep clone of the data

Reply:
With regards to the above comment @Daniel-Searle I appreciate what you are saying, but it is currently working without issue. We can absolutely add something to backlog so that when the team have agreed a consistent approach to testing across the services, we can then ensure there is consistency across them.